### PR TITLE
Add golint to the linters

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

Adds `golint` to the linters, and fix the code with it :angel: 

Related to tektoncd/plumbing#64

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: pkg/reconciler/timeout_handler.go [hunk 1]
```diff
@@ -45,8 +45,8 @@ type StatusKey interface {
 	GetRunKey() string
 }
 
-// backoff contains state of exponential backoff for a given StatusKey
-type backoff struct {
```
Comment (ImJasonH) [inline pkg/reconciler/timeout_handler.go]: Was this exported because `GetBackoff` is exported below? Can we unexport `GetBackoff` instead?

Comment (vdemeester) [reply pkg/reconciler/timeout_handler.go]: @ImJasonH I should have looked :angel: let's see :stuck_out_tongue:

Comment (vdemeester) [reply pkg/reconciler/timeout_handler.go]: We are using `GetBackoff` in the reconcilier :thinking:

File: pkg/credentials/gitcreds/ssh.go [hunk 2]
```diff
@@ -143,7 +143,7 @@ func (be *sshEntry) Write(sshDir string) error {
 	return ioutil.WriteFile(be.path(sshDir), []byte(be.privateKey), 0600)
 }
 
-func newSshEntry(u, secretName string) (*sshEntry, error) {
+func newSSHEnrty(u, secretName string) (*sshEntry, error) {
```
Comment (sbwsg) [inline pkg/credentials/gitcreds/ssh.go]: `Enrty` -> `Entry` (also above on line 66)

Comment (vdemeester) [reply pkg/credentials/gitcreds/ssh.go]: :man_facepalming:

File: pkg/reconciler/taskrun/taskrun_test.go [hunk 3]
```diff
@@ -246,7 +246,7 @@ func getTaskRunController(t *testing.T, d test.Data) (test.TestAssets, func()) {
 	ctx = cloudevent.WithClient(ctx, &cloudEventClientBehaviour)
 	c, _ := test.SeedTestData(t, ctx, d)
 	configMapWatcher := configmap.NewInformedWatcher(c.Kube, system.GetNamespace())
-	return test.TestAssets{
+	return test.Assets{
```
Comment (bobcatfish) [inline pkg/reconciler/taskrun/taskrun_test.go]: yaaaassss

**Pull request merged**